src/deno_transport.ts

```diff
@@ -192,49 +173,41 @@ export class DenoTransport implements Transport {
     this._closed(reason).then().catch();
   }
 
-  private enqueue(frame: Uint8Array): Promise<void> {
-    if (this.done) {
-      return Promise.resolve();
+  maybeWriteFrame(d?: Deferred<void>): void {
+    if (this.pendingWrite) {
+      return;
     }
-    const d = deferred<void>();
-    this.sendQueue.push({ frame, d });
-    if (this.sendQueue.length === 1) {
-      this.dequeue();
-    }
-    return d;
-  }
 
-  private dequeue(): void {
-    const [entry] = this.sendQueue;
-    if (!entry) return;
-    if (this.done) return;
-    const { frame, d } = entry;
-    write(frame, this.writer)
-      .then(() => {
-        if (this.options.debug) {
-          console.info(`< ${render(frame)}`);
-        }
-        d.resolve();
-      })
-      .catch((err) => {
-        if (this.options.debug) {
-          console.error(`!!! ${render(frame)}: ${err}`);
-        }
-        d.reject(err);
-      })
-      .finally(() => {
-        this.sendQueue.shift();
-        this.dequeue();
+    const frame = this.frames.shift();
+    if (!frame) {
+      return;
+    }
+    this.pendingWrite = writeAll(this.conn, frame).then(() => {
+      this.pendingWrite = null;
+      queueMicrotask(() => {
```

Comment:
This didn't have an observable effect so unsure if we should be keeping it.

```suggestion
      queueMicrotask(() => {
```